src/script/miniscript.h

```diff
@@ -316,11 +316,22 @@ struct MaxInt {
         return a.value + b.value;
     }
 
+    friend MaxInt<I> operator-(const MaxInt<I>& a, const I& b) {
+        if (!a.valid) return {};
+        return a.value - b;
```

Comment:
I suppose it never happens, but worth checking that `a.value >= b`?
Especially since it could be used with unsigned types.

Reply:
Hmm that's a good point re unsigned type, but it needs to be able to return negative values. So i guess it's up to the caller to make sure they don't call this with `b > a` if `a` is unsigned, as with regular substractions?